tools/python_api/src_cpp/cached_import/py_cached_import.cpp

```diff
@@ -0,0 +1,18 @@
+#include "cached_import/py_cached_import.h"
+
+namespace kuzu {
+
+PythonCachedImport::~PythonCachedImport() {
+ 	py::gil_scoped_acquire acquire;
+	allObjects.clear();
+}
+
+py::handle PythonCachedImport::addToCache(py::object obj) {
+    auto ptr = obj.ptr();
```

Comment:
Do we have to hold the gil when we add an object to cache?

Reply:
yep; though this function is only called from another function that asserts that the GIL is held, so adding another gil_scoped_acquire here would be redundant.